src/App.tsx

```diff
@@ -10,37 +12,23 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
-  return (
-    <div className="section">
-      <h1 className="title">Selected tab is Tab 1</h1>
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
 
-      <div data-cy="TabsComponent">
-        <div className="tabs is-boxed">
-          <ul>
-            <li className="is-active" data-cy="Tab">
-              <a href="#tab-1" data-cy="TabLink">
-                Tab 1
-              </a>
-            </li>
+  const selectedTab = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
 
-            <li data-cy="Tab">
-              <a href="#tab-2" data-cy="TabLink">
-                Tab 2
-              </a>
-            </li>
+  const handleChange = (tab: Tab) => {
```

Comment:
You are creating a handler (which is generally a good idea), but then there is some logic at the 'onClick' function. 
If you create handlers => handlers should cover ALL logic. 
In there case honestly there is no need to create a handler, just use `setSelectedTab` instead.